Add warning for transfer larger than CCTP max transfer limit (1M)

A user hit a limit of 100k on the CCTP bridge in Connect, we know Circle imposes limits of 1M on CCTP transfers (dependent on the chain) but are unclear on where this 100k limit is being imposed. We need to investigate where its propagating from (doesn't seem to be relayers). My hypothesis is that the CCTP limits are on an address basis (we hit CCTP from the WH CCTP contracts) and over some small window of time, amounting perhaps to 1M in the period this user tried it.

1. find the source of the limit and remedy it or decide to keep it if its unsolvable or legitimate 
2. implement elegant error handling that matches the UI of "amount must be greater than 0" below:  
<img width="785" alt="Screenshot 2024-01-19 at 4 53 10 AM" src="https://github.com/wormhole-foundation/wormhole-connect/assets/4761680/a8ccdb14-9072-4f0d-89ae-46076277f9e2">

<img width="1644" alt="Screenshot 2024-01-19 at 11 55 27 AM" src="https://github.com/wormhole-foundation/wormhole-connect/assets/689351/a9df6729-a861-436f-9625-fa8b5e5b23e3">

A test transfer of 1000005 USDC failed with the above error:

> main.js:691 Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted: Burn amount exceeds per tx limit", method="estimateGas", transaction={"from":"0x8F26A0025dcCc6Cfc07A7d38756280a10E295ad7","to":"0x774a70Bbd03327c21460B60f25b677d9e46AB458","data":"0x59b87d8e0000000000000000000000005425890298aed601595a70ab815c96711a31bc65000000000000000000000000000000000000000000000000000000e8d4f15b40000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000050000000000000000000000008f26a0025dccc6cfc07a7d38756280a10e295ad7","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":3,"message":"execution reverted: Burn amount exceeds per tx limit","data":"0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000204275726e20616d6f756e74206578636565647320706572207478206c696d6974","cause":null}}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)

--- a/wormhole-connect/src/consts/index.ts
+++ b/wormhole-connect/src/consts/index.ts
@@ -1,2 +1,4 @@
 export const GOVERNOR_WHITEPAPER_URL =
   'https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0007_governor.md';
+
+export const CCTP_MAX_TRANSFER_LIMIT = 1000000;

--- a/wormhole-connect/src/utils/transferValidation.ts
+++ b/wormhole-connect/src/utils/transferValidation.ts
@@ -112,7 +112,9 @@ export const validateAmount = (
     if (numAmount > b) return 'Amount cannot exceed balance';
   }
   if (isCctp && numAmount >= CCTP_MAX_TRANSFER_LIMIT)
-    return 'Your transaction exceeds the maximum transfer limit of 1,000,000 USDC. Please use a different amount.';
+    return `Your transaction exceeds the maximum transfer limit of ${Intl.NumberFormat(
+      'en-EN',
+    ).format(CCTP_MAX_TRANSFER_LIMIT)} USDC. Please use a different amount.`;
   if (numAmount > maxAmount) {
     return `At the moment, amount cannot exceed ${maxAmount}`;
   }

--- a/wormhole-connect/src/utils/transferValidation.ts
+++ b/wormhole-connect/src/utils/transferValidation.ts
@@ -112,7 +112,7 @@ export const validateAmount = (
     if (numAmount > b) return 'Amount cannot exceed balance';
   }
   if (isCctp && numAmount >= CCTP_MAX_TRANSFER_LIMIT)
-    return 'Burn amount exceeds per transaction limit';
+    return 'Your transaction exceeds the maximum transfer limit of 1,000,000 USDC. Please use a different amount.';
   if (numAmount > maxAmount) {
     return `At the moment, amount cannot exceed ${maxAmount}`;
   }

--- a/wormhole-connect/src/utils/transferValidation.ts
+++ b/wormhole-connect/src/utils/transferValidation.ts
@@ -22,6 +22,8 @@ import { useDebounce } from 'use-debounce';
 import { isPorticoRoute } from 'routes/porticoBridge/utils';
 import { PorticoBridgeState } from 'store/porticoBridge';
 import { DataWrapper } from 'store/helpers';
+import { CCTPManual_CHAINS as CCTP_CHAINS } from 'routes/cctpManual';
+import { CCTP_MAX_TRANSFER_LIMIT } from 'consts';
 
 export const validateFromChain = (
   chain: ChainName | undefined,
@@ -99,6 +101,7 @@ export const validateAmount = (
   amount: string,
   balance: string | null,
   maxAmount: number,
+  isCctp?: boolean,
 ): ValidationErr => {
   if (amount === '') return '';
   const numAmount = Number.parseFloat(amount);
@@ -108,6 +111,8 @@ export const validateAmount = (
     const b = Number.parseFloat(balance);
     if (numAmount > b) return 'Amount cannot exceed balance';
   }
+  if (isCctp && numAmount >= CCTP_MAX_TRANSFER_LIMIT)
+    return 'Burn amount exceeds per transaction limit';
   if (numAmount > maxAmount) {
     return `At the moment, amount cannot exceed ${maxAmount}`;
   }
@@ -231,6 +236,26 @@ export const getIsAutomatic = (route: Route | undefined): boolean => {
   return r.AUTOMATIC_DEPOSIT;
 };
 
+export const isCctp = (
+  token: string,
+  destToken: string,
+  toChain: ChainName | undefined,
+  fromChain: ChainName | undefined,
+): boolean => {
+  const isUSDCToken =
+    TOKENS[token]?.symbol === 'USDC' &&
+    TOKENS[destToken]?.symbol === 'USDC' &&
+    TOKENS[token]?.nativeChain === fromChain &&
+    TOKENS[destToken]?.nativeChain === toChain;
+  const bothChainsSupportCCTP =
+    !!toChain &&
+    CCTP_CHAINS.includes(toChain) &&
+    !!fromChain &&
+    CCTP_CHAINS.includes(fromChain);
+
+  return isUSDCToken && bothChainsSupportCCTP;
+};
+
 export const validateAll = async (
   transferData: TransferInputState,
   relayData: RelayState,
@@ -263,6 +288,7 @@ export const validateAll = async (
   const availableRoutes = routeStates
     ?.filter((rs) => rs.supported)
     .map((val) => val.name);
+  const isCctpTx = isCctp(token, destToken, toChain, fromChain);
   const baseValidations = {
     sendingWallet: await validateWallet(sending, fromChain),
     receivingWallet: await validateWallet(receiving, toChain),
@@ -270,7 +296,12 @@ export const validateAll = async (
     toChain: validateToChain(toChain, fromChain),
     token: validateToken(token, fromChain),
     destToken: validateDestToken(destToken, toChain, supportedDestTokens),
-    amount: validateAmount(amount, sendingTokenBalance, maxSendAmount),
+    amount: validateAmount(
+      amount,
+      sendingTokenBalance,
+      maxSendAmount,
+      isCctpTx,
+    ),
     route: validateRoute(route, availableRoutes),
     toNativeToken: '',
     foreignAsset: validateForeignAsset(foreignAsset),

